embedded-workshop-book/src/gdb.md

```diff
@@ -0,0 +1,79 @@
+# Using GDB
+
+To debug embedded Rust applications with GDB we currently recommend using tooling like OpenOCD,
+JLinkGDBServer or pyOCD
+
+Although `cargo-embed` v0.10 (and v0.10 of `probe-rs`, the library that powers `cargo-embed`)
+support spawning a GDB server it has some limitations
+- stepping through the code (e.g. GDB's `step` and `next` commands) is imprecise or doesn't work in
+  some cases
+- it's not possible to have a GDB server and RTT channels running at the same time (this limitation
+  is likely to be removed in v0.11)
+
+The rest of this section covers how to debug an embedded application within VS code using OpenOCD as
+the GDB server.
+
+## Dependencies
+
+1. You'll need to install OpenOCD. Installation instructions vary depending on your OS.
+
+2. Install the cortex-debug extension in VS code. Head to the extensions marketplace and search for
+   "cortex-debug". Latest version at the time of writing is 0.3.7 and the author of the extension is
+   "marus25"
+
+## Preparation
+
+For the best debugging experience, the `dev` (development) compilation profile should be set to its
+default settings. Modify `beginner/apps/Cargo.toml` to revert the `dev` profile to its default.
+
+``` diff
+ panic-log = { path = "../../common/panic-log" }
+
+ # optimize code in both profiles
+-[profile.dev]
+-codegen-units = 1
+-debug = 1
+-debug-assertions = true # !
+-incremental = false
+-lto = "fat"
+-opt-level = 'z' # !
+-overflow-checks = false
+
+ [profile.release]
+```
+
+## How to
+
```

Comment:
0. Make sure you've connected your Development Kit

Reply:
Indeed!

Reply:
(it seems trivial but there was some confusion about what's running on the target and what's running on the host in some workshops so that was the motivation behind this step.)